src/psyclone/gocean1p0.py

```diff
@@ -471,14 +471,19 @@ class GOLoop(Loop):
     '''
     _bounds_lookup = {}
 
-    def __init__(self, parent=None,
-                 topology_name="", loop_type=""):
+    def __init__(self, parent=None, topology_name="", loop_type=""):
         # pylint: disable=unused-argument
         const = GOceanConstants()
         Loop.__init__(self, parent=parent,
                       valid_loop_types=const.VALID_LOOP_TYPES)
         self.loop_type = loop_type
 
+        # Check that the GOLoop in inside the GOcean PSy-layer
+        if not self.ancestor(GOInvokeSchedule):
+            raise GenerationError(
```

Comment:
Please document this :raises: in the docstring.

Reply:
Done